lib/plug/router.ex

```diff
@@ -377,8 +377,17 @@ defmodule Plug.Router do
         raise ArgumentError, message: "expected :to to be an alias or an atom"
       end
 
+      {target, target_opts} =
+        case Code.ensure_loaded(target) do
```

Comment:
I would prefer this check:

```elixir
case Atom.to_string(target) do
  "Elixir." <> _ -> {target, target.init(plug_options)}
  _ ->  {{__MODULE__, target}, plug_options}
end
```

That's because we can have false positives. For example, there is a module named `:user` in Erlang and we would then try to call it as a module.